File: lib/models/user_profile.dart
class UserProfile {
  const UserProfile({
    required this.name, required this.age, required this.sex, required this.heightCm,
    required this.weightKg, required this.healthConditions, required this.activityLevel,
    required this.primaryGoal, required this.dietaryPreference, required this.allergies,
  });

  final String name;
  final int age;
  final String sex;
  final double heightCm;
  final double weightKg;
  final List<String> healthConditions;
  final String activityLevel;
  final String primaryGoal;
  final String dietaryPreference;
  final List<String> allergies;

  Map<String, dynamic> toJson() => {
    'basicInfo': {'name': name, 'age': age, 'sex': sex, 'heightCm': heightCm, 'weightKg': weightKg},
    'healthConditions': healthConditions,
    'lifestyle': {'activityLevel': activityLevel},
    'goals': {'primary': primaryGoal},
    'diet': {'preference': dietaryPreference, 'allergies': allergies},
  };

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    final basicInfo = Map<String, dynamic>.from(json['basicInfo'] as Map);
    final lifestyle = Map<String, dynamic>.from(json['lifestyle'] as Map);
    final goals = Map<String, dynamic>.from(json['goals'] as Map);
    final diet = Map<String, dynamic>.from(json['diet'] as Map);
    return UserProfile(
      name: basicInfo['name'] as String,
      age: (basicInfo['age'] as num).toInt(),
      sex: basicInfo['sex'] as String,
      heightCm: (basicInfo['heightCm'] as num).toDouble(),
      weightKg: (basicInfo['weightKg'] as num).toDouble(),
      healthConditions: List<String>.from(json['healthConditions'] as List),
      activityLevel: lifestyle['activityLevel'] as String,
      primaryGoal: goals['primary'] as String,
      dietaryPreference: diet['preference'] as String,
      allergies: List<String>.from(diet['allergies'] as List),
    );
  }
}

File: backend/models/UserProfile.js
const mongoose = require('mongoose');

const ACTIVITY_LEVELS = ['Sedentary', 'Lightly Active', 'Moderately Active', 'Very Active'];
const GOALS = ['Weight Loss', 'Weight Maintenance', 'Weight Gain', 'Muscle Gain', 'General Healthy Eating'];
const DIETARY_PREFERENCES = ['Vegetarian', 'Non-Vegetarian', 'Vegan', 'Eggetarian', 'Other'];
const ALLERGIES = ['Peanut', 'Milk', 'Egg', 'Soy', 'Gluten', 'Other'];
const SEXES = ['Female', 'Male', 'Intersex', 'Prefer not to say'];

const userProfileSchema = new mongoose.Schema({
  userId: { type: String, required: true, unique: true, index: true, trim: true },
  basicInfo: {
    name: { type: String, required: true, trim: true, maxlength: 100 },
    age: { type: Number, required: true, min: 1, max: 120 },
    sex: { type: String, required: true, enum: SEXES },
    heightCm: { type: Number, required: true, min: 40, max: 300 },
    weightKg: { type: Number, required: true, min: 2, max: 500 },
  },
  healthConditions: { type: [String],default: [],},
  lifestyle: { activityLevel: { type: String, required: true, enum: ACTIVITY_LEVELS } },
  goals: { primary: { type: String, required: true, enum: GOALS } },
  diet: {
    preference: { type: String, required: true, enum: DIETARY_PREFERENCES },
    allergies: { type: [String], enum: ALLERGIES, default: [] },
  },
}, { timestamps: true, versionKey: false, strict: 'throw' });

const UserProfile = mongoose.model('UserProfile', userProfileSchema);
module.exports = UserProfile;
module.exports.options = { ACTIVITY_LEVELS, GOALS, DIETARY_PREFERENCES, ALLERGIES, SEXES };
